### ISSUE TITLE
False positive for `Lint/MixedRegexpCaptureTypes`

### ISSUE THREAD

**Issue by marocchino** [rubocop/rubocop#12542]
when `>` is included in the look-ahead matcher rubocop recognizes the matcher as a named matcher.

## Actual behavior

```ruby
$ echo '/(?<=>)(<br>)(?=><)/' | bundle exec rubocop --stdin test.rb --only Lint/MixedRegexpCaptureTypes
Inspecting 1 file
W

Offenses:

test.rb:1:1: W: Lint/MixedRegexpCaptureTypes: Do not mix named captures and numbered captures in a Regexp literal.
/(?<=>)(<br>)(?=><)/
^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

## Expected behavior

No offence

```rb
  it 'does not register offense to a regexp with look behind' do
    expect_no_offenses(<<~RUBY)
      /(?<=>)(<br>)(?=><)/
    RUBY
  end
```

## Steps to reproduce the problem

```sh
echo '/(?<=>)(<br>)(?=><)/' | bundle exec rubocop --stdin test.rb --only Lint/MixedRegexpCaptureTypes
```

The following modification will fix this issue, but it cause `Metrics/CyclomaticComplexity`.
I think a more fundamental solution would be to prevent it from being recognized as a named capture.

```
diff --git a/lib/rubocop/ext/regexp_node.rb b/lib/rubocop/ext/regexp_node.rb
index 8872f849c..6985ded5e 100644
--- a/lib/rubocop/ext/regexp_node.rb
+++ b/lib/rubocop/ext/regexp_node.rb
@@ -56,6 +56,7 @@ module RuboCop
         parsed_tree&.traverse do |event, exp, _index|
           yield(exp) if event == :enter &&
                         named == exp.respond_to?(:name) &&
+                        !exp.text.start_with?('(?<=') &&
                         exp.respond_to?(:capturing?) &&
                         exp.capturing?
         end
```

## RuboCop version

```
$ bundle exec rubocop -V
1.59.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 3.2.2) [arm64-darwin22]
  - rubocop-performance 1.19.1
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.25.0
```

**Comment by KarlHeitmann** [comment]
Just an idea in order to get rid of the `Metrics/CyclomaticComplexity` offense, from what I understood on the documentation of [Metrics/CyclomaticComplexity](https://rubydoc.info/gems/rubocop/RuboCop/Cop/Metrics/CyclomaticComplexity), there is an algorithm that counts the decision made in the method. The `&&` operator adds `+1` to the score. Maybe if you add a private method that contains all the conditions in the if:

```ruby

yield(exp) if exp_conditions(exp, other_arguments)

# ...

def exp_conditions(exp, other_arguments)
  event == :enter &&
  named == exp.respond_to?(:name) &&
  !exp.text.start_with?('(?<=') &&
  exp.respond_to?(:capturing?) &&
  exp.capturing?
end

```

**Comment by marocchino** [comment]
Thanks for the suggestion. I think it's a workaround, but it looks easy, I'll create a PR.
